parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -55,11 +56,43 @@ object Tokens extends CrfTokenable {
   override def normalise(tokens: Array[String]): Array[String] = tokens.map(_.toUpperCase)
 
   def tokensToMap(tokens: Seq[CrfTokenResult]): Map[String, String] = {
-    tokens.groupBy(_.label).map {
+    val validatedTokens = validatePostCode(tokens.groupBy(_.label))
+    validatedTokens.map {
       case (token, values) => (token, values.map(_.value).mkString(" "))
     }
   }
 
+  def validatePostCode(tokens: Map[String, Seq[CrfTokenResult]]): Map[String, Seq[CrfTokenResult]] = {
+    val postCodeTokens = tokens.getOrElse(Tokens.postcode, Seq.empty)
+    if(postCodeTokens.size == 1) {
+      val token = postCodeTokens.head.value
+      if(token.length >= 5) {
+        val outcode = token.substring(token.length - 3, token.length)
+        val incode = token.substring(0, token.indexOf(outcode))
+        val regex = "[0-9][A-Z][A-Z]".r
```

Comment:
In the future we may wish to consider using
` [0-9][A-Z-[CIKMOV]]{2}`
or something more robust, but for time being this is fine.